### DISCUSSION TITLE
lock list - cursor value

### DISCUSSION THREAD

**Discussion by hehe7007** [git-lfs/git-lfs#4631] [Q&A] [answered]
Hello. I am developing lfs using git-lfs. Thank you for providing.

There is an issue because the **cursor** value does not come during development of the lfs lock list function.
What git command should I run to know if the cursor value is coming?
For example,
I can get **git lfs locks -l 100 ==> limit** URI query .
Like this, what option value do I have to give to get the **cursor**??
If I need to make other settings to get the cursor value, please share.

https://github.com/git-lfs/git-lfs/blob/main/docs/api/locking.md
Refer to that document.

thank you.

**Comment by bk2204** [comment, marked as answer]
Hey,

I'm not sure I understand completely what you're asking, but let me try to answer and you can follow up if you have questions.

When you're working with the list locks API, the first time you make the call, you don't pass the `cursor` value.  That tells the API that you want to start from the beginning.  When you get a response, if there are more than `limit` responses, then the API will return the `next_cursor` entry to indicate to you that there are more responses.  Then, when you call that API again to get additional responses, you pass the value you got for `next_cursor` as `cursor`, and the API will continue starting from the new point.  If there is no `next_cursor` entry, then you've seen all the results.

If you're developing the server side instead of the client side, then that's the behavior you'll see from the client.  Typically you'll provide a database ID for the cursor value, but you are free to do something else if you like.

**Reply by hehe7007** [reply]
Thank you for your detailed reply.
I was curious about the **git command.**

For example,
Git command: `git lfs locks` ===> (API: /info/lfs/locks?refspec=refs%2Fheads%2Fmain )
test1.txt hehe7007 ID:1
test2.txt hehe7007 ID:2
test3.txt hehe7007 ID:3
test4.txt hehe7007 ID:4

Git command: `git lfs locks -l 2` ===> ( API: /info/lfs/locks?limit=2&refspec=refs%2Fheads%2Fmain )
test1.txt hehe7007 ID:1
test2.txt hehe7007 ID:2

**I want to get the following result through the git command.**
**test3.txt hehe7007 ID:3
test4.txt hehe7007 ID:4**

I understand that if I pass the next_cursor value to the cursor, it starts from that point.
I have completed the implementation of all API servers, and I also confirmed that when a cursor is passed, the API response value starts from that point.

However, **it is difficult to check the desired result in the git command**. I am wondering how to pass cursor value to git command.
That is, I am curious about the **Git command** to be called ( API: /info/lfs/locks?limit=2&**cursor=3**&refspec=refs%2Fheads%2Fmain ).

Thanks for reading the long question

**Reply by bk2204** [reply]
The `git-lfs` binary doesn't expose the cursor functionality.  If you don't specify a limit, it will list all of the locks, invoking the API as many times as necessary using the cursor functionality under the hood, but it isn't exposed in the command-line interface.  It's not designed to be a user-visible artifact of how the command operates.

It sounds like maybe you're trying to implement some sort of user-visible API by wrapping the Git LFS client binary, and I would just advise that in general that is not a good idea.  An API generally needs to complete or fail within certain time limits, and that's not going to be the case with the binary, which may end up performing complex retries and take an indeterminate amount of time.
